### DISCUSSION TITLE
npm shell to directly execute commands using npm

### DISCUSSION THREAD

**Discussion by tijmenvangulik** [npm/feedback#136] [CLI]
I am not sure if this is a new wish or something that already exists but I could not find a npm feature or plugin which does do the following: 

The problem that I am often facing is that I have multiple projects with different tool sets. The nice thing of npm is that all commands are executed relatively from npm and not from the global space. For example I have installed cordova 9 in my project. But in the globals space I have installed cordova 10. When I am in the root directory of my project and I run cordova I get the global installed cordova (10). For example 
 cordova plugin add insomnia 

 But I actually need cordova (9) which is installed for that project. So I do a work around , I add a script to the the project with my extra command. For example:
  "scripts": { "runlocal": "cordova plugin add insomnia " }
and then I run npm run runlocal
Now it is correctly run using cordova 9 which was installed in my package.

It would be much better if npm had a simple command line where I could directly run my commands. Which would start with for example npm cli. You will get a new command prompt and every thing will be executed in the same context as npm has. 
npm cli
> cordova plugin add insomnia 
> exit

(internally it would do the same as the scripts function does do, so the feature has nothing more to do then routing every entered line to the same function which runs the scripts.)

I hope that this feature can be added to npm or that some one can point where this feature is already implemented.

**Comment by ljharb** [comment]
`npx cordova` will run the local version if installed, and `npm exec` will start a subshell in npm 7 latest with the local binaries in the PATH.

**Comment by tijmenvangulik** [comment]
Thank you for the quick reply, I am still using 6.9.0. So I will upgrade and check this feature.

**Reply by tijmenvangulik** [reply]
Ah 7 it is still in development, so that is why I could not find it

**Reply by ljharb** [reply]
`npm install -g npm@7`, and it's included by default in node 15.

**Comment by tijmenvangulik** [comment]
thanks
